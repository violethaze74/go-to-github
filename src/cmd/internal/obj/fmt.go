/*
 * The authors of this software are Rob Pike and Ken Thompson.
 *              Copyright (c) 2002 by Lucent Technologies.
 * Permission to use, copy, modify, and distribute this software for any
 * purpose without fee is hereby granted, provided that this entire notice
 * is included in all copies of any software which is or includes a copy
 * or modification of this software and in all copies of the supporting
 * documentation for such software.
 * THIS SOFTWARE IS BEING PROVIDED "AS IS", WITHOUT ANY EXPRESS OR IMPLIED
 * WARRANTY.  IN PARTICULAR, NEITHER THE AUTHORS NOR LUCENT TECHNOLOGIES MAKE ANY
 * REPRESENTATION OR WARRANTY OF ANY KIND CONCERNING THE MERCHANTABILITY
 * OF THIS SOFTWARE OR ITS FITNESS FOR ANY PARTICULAR PURPOSE.
 */

package obj

const (
	FmtWidth    = 1
	FmtLeft     = FmtWidth << 1
	FmtPrec     = FmtLeft << 1
	FmtSharp    = FmtPrec << 1
	FmtSpace    = FmtSharp << 1
	FmtSign     = FmtSpace << 1
	FmtApost    = FmtSign << 1
	FmtZero     = FmtApost << 1
	FmtUnsigned = FmtZero << 1
	FmtShort    = FmtUnsigned << 1
	FmtLong     = FmtShort << 1
	FmtVLong    = FmtLong << 1
	FmtComma    = FmtVLong << 1
	FmtByte     = FmtComma << 1
	FmtLDouble  = FmtByte << 1
	FmtFlag     = FmtLDouble << 1
)
