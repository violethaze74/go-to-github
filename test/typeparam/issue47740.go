// run -gcflags=-G=3

// Copyright 2021 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package main

import "fmt"

type Exp[Ty any] interface {
	Eval() Ty
}

type Lit[Ty any] Ty

func (lit Lit[Ty]) Eval() Ty       { return Ty(lit) }
func (lit Lit[Ty]) String() string { return fmt.Sprintf("(lit %v)", Ty(lit)) }

type Eq[Ty any] struct {
	a Exp[Ty]
	b Exp[Ty]
}

func (e Eq[Ty]) String() string {
	return fmt.Sprintf("(eq %v %v)", e.a, e.b)
}

var (
	e0 = Eq[int]{Lit[int](128), Lit[int](64)}
	e1 = Eq[bool]{Lit[bool](true), Lit[bool](true)}
)

func main() {
	fmt.Printf("%v\n", e0)
	fmt.Printf("%v\n", e1)
}
