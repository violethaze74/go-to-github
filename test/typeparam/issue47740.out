(eq (lit 128) (lit 64))
(eq (lit true) (lit true))
